src/test/lib/logging.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/lib/logging.h>
+
+#include <logging.h>
+#include <noui.h>
+#include <tinyformat.h>
+#include <util/memory.h>
+
+#include <stdexcept>
+
+DebugLogHelper::DebugLogHelper(std::string message)
+    : m_message{std::move(message)}
+{
+    m_print_connection = LogInstance().PushBackCallback(
+        [this](const std::string& s) {
+            if (m_found) return;
+            m_found = s.find(m_message) != std::string::npos;
+        });
+    noui_test_redirect();
+}
+
+void DebugLogHelper::check_found()
+{
+    noui_reconnect();
+    LogInstance().DeleteCallback(m_print_connection);
+    if (!m_found) {
+        throw std::runtime_error(strprintf("'%s' not found in debug log\n", m_message));
```

Comment:
Haven't looked closely, but I'm kind of surprised you can get away with throwing an exception in a destructor (`check_found` is called by `~DebugLogHelper`).

If this works, it seems fine. But in case there are issues, an alternative could be to use a callback instead of a scope, so instead of:

```c++
{
    ASSERT_DEBUG_LOG("Please check that your computer's date and time are correct!");
    MultiAddTimeData(1, DEFAULT_MAX_TIME_ADJUSTMENT + 1); //filter size 5
}
```

Something like:

```c++
AssertDebugLog("Please check that your computer's date and time are correct!", [&] {
    MultiAddTimeData(1, DEFAULT_MAX_TIME_ADJUSTMENT + 1); //filter size 5
});
```

Reply:
I believe C++11 makes destructors implicitly `noexcept`, which means that an uncaught exception from results in `std::terminate()` being called, which is what we want from throwing a runtime exception anyway

Reply:
I forgot about this, but indeed the test seem to fail either way, which is what we want.